src/common/util/firebase.ts

```diff
@@ -138,19 +142,20 @@ export function saveGame(game: w.SavedGame): firebase.database.ThenableReference
 
 // Cards and text
 
-export function listenToRecentCards(callback: (data: any) => any, uid?: string): void {
-  fb.database()
-    .ref(isNil(uid) ? 'recentCards' : `users/${uid}/cards`)
-    .orderByChild('timestamp')
-    .limitToLast(50)
-    .on('value', (snapshot: firebase.database.DataSnapshot) => {
-      if (snapshot) {
-        const unnormalizedCards = snapshot.val();
-        const normalizedCards = mapValues(unnormalizedCards, (card: any) => normalizeCard(card));
+/** Returns either all cards for a given user or the most recent cards belonging to any user. */
+export function listenToCards(callback: (data: any) => any, uid: string | null): void {
```

Comment:
Any reason not to return a Promise here instead of using a callback?

Reply:
The problem is that we want the callback called whenever the card list changes to keep the redux state in sync with firebase, so a single promise wouldn't work here.